cmd/scale.go

```diff
@@ -229,8 +229,8 @@ func (sc *scaleCmd) run(cmd *cobra.Command, args []string) error {
 				continue
 			}
 
-			osPublisher := *vm.StorageProfile.ImageReference.Publisher
-			if strings.EqualFold(osPublisher, "MicrosoftWindowsServer") {
+			osPublisher := vm.StorageProfile.ImageReference.Publisher
+			if osPublisher != nil && vm.StorageProfile.ImageReference != nil && strings.EqualFold(*osPublisher, "MicrosoftWindowsServer") {
```

Comment:
Shouldn't the check for `vm.StorageProfile.ImageReference != nil` be done before assigning `osPublisher := vm.StorageProfile.ImageReference.Publisher ` here?

Reply:
thanks - typo - fixed